Dockerfile

```diff
@@ -28,6 +28,7 @@ RUN dnf -y install \
     --setopt install_weak_deps=0 \
     --nodocs \
     gcc \
+    cargo \
```

Comment:
nitpick: I know it's my suggestion (I was just prototyping quickly), maybe just add a short commentary here that this is not a direct build dependency, instead we just want to copy the binary over in the next stage.